.github/workflows/build-sample-apps.yml

```diff
@@ -74,7 +74,8 @@ jobs:
     # CLI to replace strings in files. The CLI recommends using `cargo install` which is slow. This Action is fast because it downloads pre-built binaries. 
     # If using sd on macos, "brew install" works great. for Linux, this is the recommended way. 
     - name: Install sd CLI to use later in the workflow 
-      uses: kenji-miyake/setup-sd@v1
+      # uses: kenji-miyake/setup-sd@v1
```

Comment:
`kenji-miyake/setup-sd@v1` doesn't seem to have many users/stars, can we look for an alternative so we don't bump into issues like these?

Reply:
Good suggestion to bring up. 

The `sd` CLI is a more stable `sed` tool. None of us like having to use `sed` in any of our codebases so I would love to not find an alternative `setup-sd` action, but find an alternative solution to no longer need `sd` at all. 

I would personally like to move to [a semantic-release plugin](https://github.com/jpoehnelt/semantic-release-replace-plugin) since updating the version string in our code is part of the deployment process. Plus, by doing string replacement via code instead of CLI, the developer experience for us will be better.

Reply:
[Ticket made](https://github.com/customerio/issues/issues/11701).